app/controllers/items_controller.rb

```diff
@@ -1,6 +1,26 @@
 class ItemsController < ApplicationController
   before_action :authenticate_user!
-  before_action :set_wishlist_and_item, only: %i(edit update)
+  before_action :set_wishlist, only: %i(new create edit update)
+  before_action :set_item, only: %i(edit update)
+
+  def new
+    @item = @wishlist.items.new
+    authorize! :new, @item
+  rescue CanCan::AccessDenied
+    redirect_to root_url, alert: t("generic_error")
```

Comment:
I'd redirect to the wishlists being edited, not root. Here and in other `redirect`s in the `ItemsController`.